inc/networkname.class.php

```diff
@@ -404,28 +408,27 @@ static function showFormForNetworkPort($networkPortID) {
       $number_names = 0;
 
       if ($networkPortID > 0) {
-         $query = "SELECT `id`
-                   FROM `".$name->getTable()."`
-                   WHERE `itemtype` = 'NetworkPort'
-                   AND `items_id` = '$networkPortID'
-                   AND `is_deleted` = 0";
-
-         $result = $DB->query($query);
-
-         if ($DB->numrows($result) > 1) {
-            echo "<tr class='tab_bg_1'><th colspan='4'>" .
-                   __("Several network names available! Go to the tab 'Network Name' to manage them.") .
-                 "</th></tr>\n";
-            return;
-         }
+         $iterator = $DB->request([
+            'SELECT' => 'id',
+            'FROM'   => $name->getTable(),
+            'WHERE'  => [
+               'itemtype'     => 'NetworkPort',
+               'items_id'     => $networkPortID,
+               'is_deleted'   => 0
+            ]
+         ]);
+         $numrows = count($iterator);
 
-         switch ($DB->numrows($result)) {
+         switch ($numrows) {
             case 1 :
-               $nameID = $DB->fetchAssoc($result);
-               $name->getFromDB($nameID['id']);
+               $result = $iterator->next();
+               $name->getFromDB($result['id']);
                break;
 
             case 0 :
+               echo "<tr class='tab_bg_1'><th colspan='4'>" .
```

Comment:
This message was when we had more than 1 result, and we had a `return;` just after.

Reply:
Indeed... I've fixed